WebUI/Pages/GameDetails.razor

```diff
@@ -48,7 +60,68 @@ else
         <a class="btn btn-success rounded mr-1" href="@downloadLink" target="_blank">
             <span class="oi oi-data-transfer-download mr-1" aria-hidden="true" /> Download
         </a>
-        <button class="btn btn-primary">Leave Review</button>
+        @if (!isGameReviewedByUser)
+        {
+            <button class="btn btn-primary" @onclick="@ToggleReviewVisibility">Leave Review</button>
+        }
+    }
+    @if (showReviewForm)
+    {
+        <br />
+        <div class="bg-white rounded shadow-sm p-4 mb-5 rating-review-select-page">
+            <h5 class="mb-4">Leave Review</h5>
+            <form>
+                <div class="form-group pb-4">
+                    <span class="rating">
+                        @for (int i = 5; i > 0; i--)
+                        {
+                            <input type="radio" name="starRating" id=@($"rating{i}") value="@i" @onchange="RatingSelectionChanged">
+                            <label for=@($"rating{i}")>@i</label>
+                        }
+                    </span>
+                </div>
+                <div class="form-group pt-4">
+                    <h6 class="mb-4">Comments:</h6>
+                    <textarea class="form-control" id="comments" @bind="description"></textarea>
+                </div>
+                <div class="form-group">
+                    <button class="btn btn-primary btn-sm" type="submit" @onclick="(() => SubmitReview())">Submit Review</button>
+                </div>
+            </form>
+        </div>
+        <br />
+    }
+    @if (approvedReviews.Count == 0)
+    {
+        <h4 class="my-4">No reviews available... ;-;</h4>
+    }
+    else
+    {
+        @foreach (var review in approvedReviews)
+        {
+            <div class="border border-secondary rounded mr-3 my-4 p-3 bg-light d-flex-column">
+                <div class="py-2 pl-2"><b>@review.User.FirstName @review.User.LastName</b></div>
+                <span class="rating">
+                    @for (int i = 5; i > 0; i--)
+                    {
+                        <input type="radio" name="starRating" id=@($"rating{i}") checked="@(review.Rating == i)" disabled>
```

Comment:
You should be very careful with radio groups. Each radio group is required to have a unique name, otherwise all radio buttons on the page will share the same group (like it happens right now if you have more than one review).
```suggestion
                        <input type="radio" name="@($"rating-{review.GameId}-{review.UserId}")" id=@($"rating{i}") checked="@(review.Rating == i)" disabled>
```

Reply:
This is also an argument towards never using radio buttons in your life ever again

Reply:
Radio buttons are the bane of my existence lol